### DISCUSSION TITLE
Icon Request: icon-vm (icon-virtual-machine)

### DISCUSSION THREAD

**Discussion by leehambley** [FortAwesome/Font-Awesome#21144] [Icon Requests]
I'd like to propose adding a VM icon to FontAwesome. There's no accepted standard, but maybe we could adopt the de-facto standard which is the icon that VMWare is using for their VMs(which I guess isn't copy right, at least I couldn't find anything.)

First of all VMWare's own logo, which **isn't** what I'm suggesting we use:

![screenshot 2014-12-04 20 32 58](https://cloud.githubusercontent.com/assets/18952/5304616/c5a45eb8-7bf4-11e4-958b-6e76a67b5bb2.png)

The widest accepted icon for what a VM looks like is apparently this:

![untitled](https://cloud.githubusercontent.com/assets/18952/5304681/4ae1b2a6-7bf5-11e4-9f0a-da8148dcc11e.png)

I think with very little work, it might be possible to add this to FontAwesome, if there's some guide to contributing (I'm capable _enough_ with vector tools) I'd be more than happy to bravely attempt a pull request.

**Comment by leehambley** [comment]
I took at shot at two svg versions, GH can't handle SVG attachments, so here they are as XML.

### SVG Sources:

``` xml
# icon-ws-overlapped
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="196px" height="206px" viewBox="0 0 196 206" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns">
    <!-- Generator: Sketch 3.2 (9961) - http://www.bohemiancoding.com/sketch -->
    <title>Slice 1</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" sketch:type="MSPage">
        <path d="M8,27.996374 C8,16.9526816 16.9617664,8 27.996374,8 L138.003626,8 C149.047318,8 158,16.9617664 158,27.996374 L158,138.003626 C158,149.047318 149.038234,158 138.003626,158 L27.996374,158 C16.9526816,158 8,149.038234 8,138.003626 L8,27.996374 Z" id="Path-1" stroke="#3B3B39" stroke-width="15" sketch:type="MSShapeGroup"></path>
        <path d="M38,67.996374 C38,56.9526816 46.9617664,48 57.996374,48 L168.003626,48 C179.047318,48 188,56.9617664 188,67.996374 L188,178.003626 C188,189.047318 179.038234,198 168.003626,198 L57.996374,198 C46.9526816,198 38,189.038234 38,178.003626 L38,67.996374 Z" id="Path-2" stroke="#3B3B39" stroke-width="15" sketch:type="MSShapeGroup"></path>
    </g>
</svg>
```

``` xml
<!-- icon-ws-stacked -->
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="201px" height="176px" viewBox="0 0 201 176" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns">
    <!-- Generator: Sketch 3.2 (9961) - http://www.bohemiancoding.com/sketch -->
    <title>Path 1 + Path 2</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" sketch:type="MSPage">
        <g id="Path-1-+-Path-2" sketch:type="MSLayerGroup" transform="translate(8.000000, 8.000000)" stroke="#3B3B39" stroke-width="15">
            <path d="M150,27.7109375 L150,20.0038457 C150,8.95563302 141.047318,0 130.003626,0 L19.996374,0 C8.96176644,0 0,8.95469235 0,20.0008652 L0,99.9991348 C0,111.054067 8.95629335,120 20.0044411,120 L22.8671875,120" id="Path-1" sketch:type="MSShapeGroup"></path>
            <path d="M35,60.0008652 C35,48.9546924 43.9617664,40 54.996374,40 L165.003626,40 C176.047318,40 185,48.9459329 185,60.0008652 L185,139.999135 C185,151.045308 176.038234,160 165.003626,160 L54.996374,160 C43.9526816,160 35,151.054067 35,139.999135 L35,60.0008652 Z" id="Path-2" sketch:type="MSShapeGroup"></path>
        </g>
    </g>
</svg>
```

### Dimensions, weights, etc:

Box width: `150`
Box height: `120`
Aspect Ratio: `5:4`
Color: `3B3B39` 
Stroke Width: 15

### PNG Previews:

![icon-ws-stacked](https://cloud.githubusercontent.com/assets/18952/5308265/6f67fa20-7c15-11e4-8b74-1c9ef69e7d4d.png)
![icon-ws-overlapping](https://cloud.githubusercontent.com/assets/18952/5308266/6f68871a-7c15-11e4-9aad-508031a297c0.png)

Personally I think the "overlapped" looks a tonne better, I'm still digging into how SVGs make it into webfonts, but maybe I can submit a PR to https://github.com/encharm/Font-Awesome-SVG-PNG/tree/master/black/svg if that is the source asset repo?

**Comment by RileyStarlight** [comment]
I'm not very fond of vmware, but I think an icon to virtual machines would be fine. And if it is generic, the better.
+1

**Comment by leehambley** [comment]
Indeed, that was the result of my research, seems like VMWare is more or less the de facto standard for paid virtual machines on the desktop, but I think this icon stands on it's own.

Do you know how I can make a PR to put this icon into FA? It's kinda poorly documented, I think even the icon sources live in another repository?

**Comment by SatanicGeek** [comment]
+1

**Comment by leegraves** [comment]
+1

**Comment by spicyramen** [comment]
+1

**Comment by yusufozturk** [comment]
+1

**Comment by yusufozturk** [comment]
+1

**Comment by tekstrong** [comment]
Would love to see a VM icon as well.

**Comment by leehambley** [comment]
~~FWIW Octicons has added this (not sure when, but I came across it today as it happens.) https://octicons.github.com/icon/terminal~~

~~By the way for the sake of the spam, let's try and use the "reactions" feature, and not post +1s as comments!~~ :metal: 

Edit, ignore me, I had two open PRs for FontAwesome, and I thought this was the one about the terminal icon, sorry for the spam. I'll leave this comment here as testament to my inability to read.

**Comment by VeloAddict** [comment]
+1

**Comment by 4b42** [comment]
+1

**Comment by Giovancruz** [comment]
+1

**Comment by imseandavis** [comment]
+1

**Comment by NickCraver** [comment]
Needs a new icon label :)

**Comment by gusbemacbe** [comment]
It is sad that this icon hasn't been release for 2 years.

**Comment by imseandavis** [comment]
@tagliala Any update on if this is still being considered?

**Comment by aiso-net** [comment]
would like to see this icon as well

**Comment by tagliala** [comment]
Does https://fontawesome.com/icons/clone?style=regular fit the use case?

**Comment by aiso-net** [comment]
I would say that would do, thanks!

**Comment by OscarBeiro** [comment]
+1

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂

**Comment by OffalGood** [comment]
+1
